core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -4992,7 +4993,7 @@ public int getBloomFilterElementCount() {
     public BloomFilter getBloomFilter(double falsePositiveRate) {
         beginBloomFilterCalculation();
         try {
-            return getBloomFilter(getBloomFilterElementCount(), falsePositiveRate, (long) (Math.random() * Long.MAX_VALUE));
+            return getBloomFilter(getBloomFilterElementCount(), falsePositiveRate, new Random().nextInt());
```

Comment:
This is where the randomness for Bloom filters for a wallet is generated.

Reply:
This, or maybe FilterMerger line 47.